File: DoohClick.Web/src/app/shared/model/response.model.ts

export interface MvGridResponse<T> {
    data?: T[];
    totalRows: number;
}

export interface MvResponse<T> {
    type: string;
    message: string;
    data?: T
}

export interface MvListitemDdl {
    id: number;
    item: string;
    code: string;
}

File: DoohClick.API/DoohClick.Model/Shared/Response/ResponseModel.cs

namespace DoohClick.Model.Shared.Response
{
    public record MvResponse<T>
    {
        public required string Type { get; set; }
        public required string Message { get; set; }
        public T? Data { get; set; }
    }
    public class MvGridResponse<T>
    {
        public List<T>? Data { get; set; }
        public int TotalRows { get; set; }
    }
}
